src/UriFactory/GuzzleFactory.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Http\Discovery\UriFactory;
+
+use GuzzleHttp\Psr7\Uri;
+use Http\Message\UriFactory;
+
+/**
+ * Creates a guzzlehttp/psr7 URI object
+ *
+ * @author David de Boer <david@ddeboer.nl>
+ */
+class GuzzleFactory implements UriFactory
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function createUri($uri)
+    {
```

Comment:
Done. By the way, if you comment on the line itself, the discussion will be cleaned up if that line changes.)